llvm/lib/Target/AIE/AIE2GenInstrInfo.td

```diff
@@ -12,6 +12,7 @@ let hasSideEffects = false, mayLoad = false, mayStore = false in {
 let Itinerary = II_ADD in {
   let Defs = [srCarry] in
   def ADD_add_r_ri : AIE2_add_r_ri_inst_alu<(outs eR:$mRx), (ins eR:$mRx0, simm7:$c7s), "add", "$mRx, $mRx0, $c7s">;
+  def ADD_add_r_ri_imm0 : AIE2_add_r_ri_inst_alu_imm0<(outs eR:$mRx), (ins eR:$mRx0), "add", "$mRx, $mRx0, 0">;
```

Comment:
This writes the carry flag. Perhaps not important now, but it prevents us to use ADC safely.
Instead you can use `rd = OR rs, rs`, which is a pure move.